x/axelarcork/keeper/proposal_handler.go

```diff
@@ -15,40 +15,57 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/peggyjv/sommelier/v7/x/axelarcork/types"
+	pubsubtypes "github.com/peggyjv/sommelier/v7/x/pubsub/types"
 )
 
+func NewAxelarSubscriptionID(chainID uint64, address common.Address) string {
+	return fmt.Sprintf("%d:%s", chainID, address.String())
+}
+
 // HandleAddManagedCellarsProposal is a handler for executing a passed community cellar addition proposal
 func HandleAddManagedCellarsProposal(ctx sdk.Context, k Keeper, p types.AddAxelarManagedCellarIDsProposal) error {
 	config, ok := k.GetChainConfigurationByID(ctx, p.ChainId)
 	if !ok {
 		return fmt.Errorf("chain by id %d not found", p.ChainId)
 	}
 
+	_, publisherFound := k.pubsubKeeper.GetPublisher(ctx, p.PublisherDomain)
+	if !publisherFound {
+		return fmt.Errorf("not an approved publisher: %s", p.PublisherDomain)
+	}
+
```

Comment:
would it be more flexible to just not create a default subscription if the publisher doesn't exist yet? steward would cache the cellar ID but not any trust data and thus it wouldn't be callable. just seems like it would avoid an annoying situation where the props get made out of order

Reply:
I expect adding a publisher to be a fairly rare event, and we can make the documentation clear. Plus when you submit the prop it will speculatively execute and fail if the publisher doesn't exist, and reject it (I believe), so out of order is an unlikely scenario. Also if we leave it optional and someone forgets, then they have to make a second prop for a default subscription.